[2.x] feat: Include `DatabaseQueue` in core

imorland opened the pull request:
<!--
IMPORTANT: We applaud pull requests, they excite us every single time. As we have an obligation to maintain a healthy code standard and quality, we take sufficient time for reviews. Please do create a separate pull request per change/issue/feature; we will ask you to split bundled pull requests.
-->

**Fixes #0000**

**Changes proposed in this pull request:**
<!-- fill this out, mention the pages and/or components which have been impacted -->

**Reviewers should focus on:**
<!-- fill this out, ask for feedback on specific changes you are unsure about -->

**Screenshot**
<!-- include an image of the most relevant user-facing change, if any -->

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)

imorland committed:
feat: switch to db queue as default

StyleCIBot committed:
Apply fixes from StyleCI

Commit:
fix: correct condition for displaying not sending message in MailPage (#4260)

flarum-bot committed:
Bundled output for commit 54fcc875d54dbd6efe055176186cfe9126b69eaa
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
[2.x] chore: make `PostUser` more extensible (#4252)

* chore: make PostUser more extensible

* fix: userOnline can be null

flarum-bot committed:
Bundled output for commit c66f9239ee6a195a931e5cd4d71624e7be625779
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

SychO9 committed:
fix: include LabelValue in the default bundle (sync) (#4235)

flarum-bot committed:
Bundled output for commit 83318e51cfeed66f92b938c7aca481ac0fb14126
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
fix: error when importing from an extension unless enabled (#4261)

flarum-bot committed:
Bundled output for commit 005706bdb4f7fd02ca100131ccc27d4053d099ee
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

luceos commented:
I honestly don't think db queue should be set to the default from the get go. It makes the hurdle to "go live" even more difficult to achieve. What we should look for in 2.x is the ability to understand community scale, then offer to enable the db queue with sufficient documentation on what to do to make it work. And yes, I do believe the db queue should be part of core, just not the default.

Flarum has set out to be as simple as possible, without much configuration. This change steers away from that.

imorland commented:
I came to a similar conclusion here, which is why this has not seen any progress..

The database queue is actually not difficult to run or maintain, it's only requirement is the scheduler being active. Therefore my thinking for when I come back to this is as follows:

- If the scheduler is not running, default to the `sync` queue (current `1.x` and `2.x` behaviour)
- If the scheduler is running, switch to the `database` queue
- Provide a mechanism via `config.php` to pin the queue to a given driver, for maximum flexibility

Additional things to consider:

- What happens when another queue driver is introduced (redis, messageMQ, etc)

I think step 1 (sync vs db) is fairly clear and straightforward to implement and document. Step 2 (other queues) still needs some thought...

DavideIadeluca committed:
fix(core): restore `xdebug.max_nesting_level` after less compilation (#4263)

flarum-bot committed:
Bundled output for commit 43f7a68f2b88f8e5f0a82bab581296f84cd2f8ef
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
fix: prevent avatar editor overlay stretching (#4265)

flarum-bot committed:
Bundled output for commit 10e769988d6e553ee0e2a1cdb1bd4645f42763b9
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
[2.x] fix: don't see developer tokens if not in the first page of results (#4266)

* fix: don't see developer tokens if not in the first page of results

* chore: phpstan

imorland committed:
chore: bump appver in prep for beta 4 (#4267)

flarum-bot committed:
Bundled output for commit 8992ecbcd085f3eb65c698f9ba4d01b1be5f1c3e
Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]
